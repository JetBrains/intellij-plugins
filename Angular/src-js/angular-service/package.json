{
  "name": "ws-typescript-angular-plugin",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "@volar/typescript": "2.4.11",
    "tsc-ide-plugin": "file:/../../../../plugins/JavaScriptLanguage/src/tscplugin",
    "esbuild": "^0.20.0"
  },
  "devDependencies": {
    "@types/node": "^18.13.10",
    "typescript": "^5.1.0"
  },
  "scripts": {
    "build:clean": "npm run clean && npm run build",
    "clean": "rm -rf ../../gen-resources/angular-service",
    "build": "tsc --noEmit && node build && cp -r ./package.json ../../gen-resources/angular-service/node_modules/ws-typescript-angular-plugin"
  }
}
