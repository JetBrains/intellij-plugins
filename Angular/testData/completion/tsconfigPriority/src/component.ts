import { ChangeDetectionStrategy, Component, OnDestroy, inject, Signal } from '@angular/core';

export interface Errors {
  [key: string]: string;
}

@Component({
  selector: 'cdt-list-errors',
  standalone: true,
  templateUrl: './component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppComponent implements OnDestroy {
  protected readonly formErrorsStore!: { errors: Signal<string[]>, setErrors: (errors: Errors) => void }

  ngOnDestroy() {
    this.formErrorsStore.setErrors({});
  }
}
