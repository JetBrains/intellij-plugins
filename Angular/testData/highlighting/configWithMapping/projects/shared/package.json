{
  "name": "shared",
  "version": "0.0.1",
  "peerDependencies": {
    "@angular/common": "^19.0.0",
    "@angular/core": "^19.0.0"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "sideEffects": false
}
