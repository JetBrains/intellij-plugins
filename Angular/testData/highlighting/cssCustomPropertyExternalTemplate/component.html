<div [style.--test2]="'#ff0000'">
</div>
