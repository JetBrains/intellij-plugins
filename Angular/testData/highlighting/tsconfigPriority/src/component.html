{{ formErrorsStore.<error descr="TS2339: Property 'errs' does not exist on type '{ errors: Signal<string[]>; setErrors: (errors: Errors) => void; }'.">errs</error>() }}
{{ formErrorsStore.errors().<error descr="TS2339: Property 'foo' does not exist on type 'string[]'.">foo</error> }}
@if (formErrorsStore.errors().length > 0) {
  <ul class="error-messages" <warning descr="Attribute fooAttr is not allowed here">fooAttr</warning>>
    @for (error of formErrorsStore.errors(); track error) {
      <li data-e2e-id="error">
        {{ error }}
        {{ <error descr="TS2339: Property 'badError' does not exist on type 'AppComponent'.">badError</error> }}
      </li>
    }
  </ul>
}
