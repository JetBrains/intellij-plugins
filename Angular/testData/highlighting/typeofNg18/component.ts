import {Component} from '@angular/core';


@Component({
             selector: 'robot-profile',
             standalone: true,
             templateUrl: "./typeof.html"
           })
export class RobotProfileComponent {
  value!: string | number

  check (value: boolean) {

  }
}
