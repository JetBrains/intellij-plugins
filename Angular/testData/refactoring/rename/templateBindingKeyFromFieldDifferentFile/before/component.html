<div *appClicks="true; param: 'foo'"></div>
<ng-template [appClicks]="true" [appClicksParam]="'foo'"></ng-template>/
