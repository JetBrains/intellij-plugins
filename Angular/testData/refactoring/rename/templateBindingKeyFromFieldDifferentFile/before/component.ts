import {Component} from '@angular/core';
import {AppClicksDirective} from "./templateBindingKeyFromFieldDifferentFile"

@Component({
    selector: 'test',
    templateUrl: './component.html',
    imports: [
        AppClicksDirective
    ]
})
export class ChipComponent {

}
