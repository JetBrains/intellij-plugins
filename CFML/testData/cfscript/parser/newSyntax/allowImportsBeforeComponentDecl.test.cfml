import "cfc_filepath1"
import 'cfc_filepath2'
import package.package.component
/* comment */
component {

}