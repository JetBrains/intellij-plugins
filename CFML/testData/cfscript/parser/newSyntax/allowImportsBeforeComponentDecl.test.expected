CfmlFile:allowImportsBeforeComponentDecl.cfml
  IMPORT_EXPRESSION
    PsiElement(IMPORT_KEYWORD)('import')
    StringLiteral
      PsiElement(DOUBLE_QUOTE)('"')
      PsiElement(STRING_TEXT)('cfc_filepath1')
      PsiElement(DOUBLE_QUOTE_CLOSER)('"')
  IMPORT_EXPRESSION
    PsiElement(IMPORT_KEYWORD)('import')
    StringLiteral
      PsiElement(SINGLE_QUOTE)(''')
      PsiElement(STRING_TEXT)('cfc_filepath2')
      PsiElement(SINGLE_QUOTE_CLOSER)(''')
  IMPORT_EXPRESSION
    PsiElement(IMPORT_KEYWORD)('import')
    COMPONENT_REFERENCE
      PsiElement(IDENTIFIER)('package')
      PsiElement(POINT)('.')
      PsiElement(IDENTIFIER)('package')
      PsiElement(POINT)('.')
      PsiElement(COMPONENT_KEYWORD)('component')
  PsiComment(COMMENT)('/* comment */')
  COMPONENT_DEFINITION
    PsiElement(COMPONENT_KEYWORD)('component')
    PsiElement(L_CURLYBRACKET)('{')
    PsiElement(R_CURLYBRACKET)('}')